native/libcst/src/parser/grammar.rs

```diff
@@ -0,0 +1,2930 @@
+                        comma_separate(sitems.0, sitems.1, Some(scomma)),
+                        comma_separate(ditems.0, ditems.1, None),
+                    )
+            }
+            / items:separated(<kwarg_or_starred()>, <comma()>) {
+                    comma_separate(items.0, items.1, None)
+            }
+            / items:separated(<kwarg_or_double_starred()>, <comma()>) {
+                    comma_separate(items.0, items.1, None)
+            }
+
+        rule kwarg_or_starred() -> Arg<'a>
+            = _kwarg()
+            / starred_expression()
+
+        rule kwarg_or_double_starred() -> Arg<'a>
+            = _kwarg()
+            / star:lit("**") e:expression() { make_star_arg(star, e) }
+
+        rule _kwarg() -> Arg<'a>
+            = n:name() eq:lit("=") v:expression() {
+                make_kwarg(n, eq, v)
+            }
+
+        rule atom() -> Expression<'a>
+            = n:name() { Expression::Name(n) }
+            / n:lit("True") { Expression::Name(make_name(n)) }
+            / n:lit("False") { Expression::Name(make_name(n)) }
+            / n:lit("None") { Expression::Name(make_name(n)) }
+            / &(tok(STRING, "") / tok(FStringStart, "")) s:strings() {s.into()}
+            / n:tok(Number, "NUMBER") { make_number(n) }
+            / &lit("(") e:(tuple() / group() / (g:genexp() {Expression::GeneratorExp(g)})) {e}
+            / &lit("[") e:(list() / listcomp()) {e}
+            / &lit("{") e:(dict() / set() / dictcomp() / setcomp()) {e}
+            / lit("...") { Expression::Ellipsis(Ellipsis {lpar: vec![], rpar: vec![]})}
+
+        rule strings() -> String<'a>
+            = s:(str:tok(STRING, "STRING") t:&_ {(make_string(str), t)}
+                / str:fstring() t:&_ {(String::Formatted(str), t)})+ {
+                make_strings(s)
+            }
+
+        rule tuple() -> Expression<'a>
+            = lpar:lpar() first:star_named_expression() &lit(",")
+                rest:(c:comma() e:star_named_expression() {(c, e)})*
+                trailing_comma:comma()? rpar:rpar() {
+                    Expression::Tuple(
+                        make_tuple(first, rest, trailing_comma, Some(lpar), Some(rpar))
+                    )
+            }
+            / lpar:lpar() rpar:lit(")") {
+                Expression::Tuple(Tuple::default().with_parens(
+                    lpar, RightParen { whitespace_before: Default::default(), rpar_tok: rpar }
+                ))}
+
+        rule group() -> Expression<'a>
+            = lpar:lpar() e:(yield_expr() / named_expression()) rpar:rpar() {
+                e.with_parens(lpar, rpar)
+            }
+
+        rule try_stmt() -> Try<'a>
+            = kw:lit("try") lit(":") b:block() f:finally_block() {
+                make_try(kw, b, vec![], None, Some(f))
+            }
+            / kw:lit("try") lit(":") b:block() ex:except_block()+ el:else_block()?
+                f:finally_block()? {
+                    make_try(kw, b, ex, el, f)
+            }
```

Comment:
[compat] for 3.11 we'll need except_star_block handling. Makes sense not to have yet